lib/resources/odata.js

```diff
@@ -64,11 +64,11 @@ module.exports = (service, endpoint) => {
           const options = QueryOptions.fromODataRequest(params, query);
           return Promise.all([
             Forms.getFields(form.def.id),
-            Submissions.streamForExport(form.id, draft, undefined, options),
-            ((params.table === 'Submissions') && options.hasPaging())
-              ? Submissions.countByFormId(form.id, draft, options) : resolve(null)
+            (((params.table === 'Submissions') && options.hasPaging())
+              ? Submissions.countByFormId(form.id, draft, options) : resolve(null)),
+            Submissions.streamForExport(form.id, draft, undefined, options) // Node16 upgrade: streaming promise has to be last in the array, I don't understand the reason.
```

Comment:
hi @issa-tseng, do you know what could be the reason of this?

Reply:
@sadiqkhoja it may be unrelated to order.  To debug could it help to unroll this array and run the queries sequentially?

Reply:
Slonik v23.9.0 is the culprit, using v23.6.0. Thanks @alxndrsn  for helping me out

Reply:
Do we know what the issue with Slonik v23.9.0 is? Some issue with streams that appears in Node 16 but not Node 14? (Or does v23.9.0 not even work in Node 14?)

Or more importantly, do we know whether the issue is fixed in later versions of Slonik? Just want to check that merging this PR wouldn't affect our ability to upgrade Slonik as part of the next release.

Reply:
I have narrowed down the issue to a commit level: https://github.com/gajus/slonik/commit/72f97820f9fe5f83a8820029d8ead64189f48360

It was released in v23.6.3, so we can't use anything latest than that. Since we have a PR for migrating postgres.js, I don't think I should invest more time in slonik

Reply:
Looking at that commit, it sort of seems like Slonik is waiting for the active query to complete before executing the next query. That would explain why shuffling the promise order had an effect: the stream won't complete until it's processed, after the `Promise.all()`, which means that the stream is blocking the query following it. The .csv.zip endpoint has multiple streams, so that seems potentially tricky.

Could it be that this is fixed in a later version of Slonik? If the explanation above is accurate, it seems like that would also be happening in Node 14. But from @alxndrsn's benchmarking, it seems like later versions of Slonik are working for us in Node 14.

Reply:
Slonik v23.6.3 doesn't work with Node 14 as well. I think issue has been fixed in later versions but those versions require other changes as well, we can upgrade Slonik in separate branch or merge postgresql.js branch